test/executor_test.rb

```diff
@@ -141,4 +141,36 @@ def test_did_close_clears_diagnostics
   ensure
     @store.delete("file:///foo.rb")
   end
+
+  def test_detects_rubocop_if_direct_dependency
+    stub_dependencies(rubocop: true, syntax_tree: false)
+    RubyLsp::Executor.new(@store, @message_queue)
+      .execute(method: "initialize", params: { initializationOptions: { formatter: "auto" } })
+    assert_equal("rubocop", @store.formatter)
+  end
+
+  def test_detects_syntax_tree_if_direct_dependency
+    stub_dependencies(rubocop: false, syntax_tree: true)
+    RubyLsp::Executor.new(@store, @message_queue)
+      .execute(method: "initialize", params: { initializationOptions: { formatter: "auto" } })
+    assert_equal("syntax_tree", @store.formatter)
+  end
+
+  def test_gives_rubocop_precedence_if_syntax_tree_also_present
+    stub_dependencies(rubocop: true, syntax_tree: true)
+    RubyLsp::Executor.new(@store, @message_queue).send(
+      :initialize_request,
+      { initializationOptions: { formatter: "auto" } },
+    )
+    assert_equal("rubocop", @store.formatter)
+  end
```

Comment:
Don't we also want to test if it sets `none` when rubocop and syntax_tree both aren't present?